redfish-core/lib/log_services.hpp

```diff
@@ -6720,27 +6720,138 @@ inline void getRedfishUriByDbusObjPath(
 }
 
 /**
- * @brief API used to get "PrettyName" by using the given dbus object path
- *        and fill into "Message" property of LogEntry schema.
+ * @brief API used to load the message and Message Args for the HW Isolation
+ * Entries
  *
+ * @param[in] ec - error code from the dbus call.
+ * @param[in] prettyName - The prettyname fetched from the dbus object.
  * @param[in] asyncResp - The redfish response to return.
- * @param[in] dbusObjPath - The DBus object path which represents redfishUri.
+ * @param[in] path - D-bus object path to find the pretty name
  * @param[in] entryJsonIdx - The json entry index to add isolated hardware
  *                            details in the appropriate entry json object.
+ * @param[in] guardType - guardType value to be used in the message
+ *
+ * @return The redfish response with "Message" property of
+ *         Hw Isolation Entry schema if success else return the error
+ */
+static void
+    loadHwIsolationMessage(const boost::system::error_code& ec,
+                           const std::string& prettyName,
+                           const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                           const std::string& path, const size_t entryJsonIdx,
+                           const std::string& guardType)
+{
+    // Determine base path based on entryJsonIdx
+    nlohmann::json::json_pointer msgPropPath;
+    nlohmann::json::json_pointer msgArgPropPath;
+    if (entryJsonIdx > 0)
+    {
+        msgPropPath = nlohmann::json::json_pointer(
+            "/Members/" + std::to_string(entryJsonIdx - 1) + "/Message");
+        msgArgPropPath = nlohmann::json::json_pointer(
+            "/Members/" + std::to_string(entryJsonIdx - 1) + "/MessageArgs");
+    }
+    else
+    {
+        msgPropPath = nlohmann::json::json_pointer("/Message");
+        msgArgPropPath = nlohmann::json::json_pointer("/MessageArgs");
+    }
+    std::vector<std::string_view> messageArgs;
+    messageArgs.push_back(guardType);
+    if (ec || prettyName.empty())
+    {
+        messageArgs.push_back(path);
+    }
+    else
+    {
+        messageArgs.push_back(prettyName);
+    }
+    const redfish::registries::Message* msgReg =
+        registries::getMessage("OpenBMC.0.6.GardRecord");
+    if (msgReg == nullptr)
+    {
+        BMCWEB_LOG_ERROR(
+            "Failed to get the GardRecord message registry to add in the condition");
+        messages::internalError(asyncResp->res);
+        return;
+    }
+    std::string message = msgReg->message;
+
+    // messageArgs.resize(msgReg->numberOfArgs);
+
+    std::string msg = redfish::registries::fillMessageArgs(messageArgs,
```

Comment:
On error an empty string will be returned by fillMessageArgs(). Should it be handled as an error?

Reply:
I see other caller of fillMessageArgs() that handles the empty message return as an internal error. I think the same should be done here: https://github.com/ibm-openbmc/bmcweb/blob/d2fe0d8eb1940375b34a0b6d5d6a5a93097d0910/redfish-core/lib/log_services.hpp#L5384

Reply:
sure. done